### PR TITLE
Add path ~/.config/conda/condarc into SEARCH_PATH to follow the XDG Base Directory Specification.

### PR DESCRIPTION
**Pull request by Hsins**
[The XDG Base Directory Specification (XDGBDS)](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) unifies the location of commonly needed kinds of files, such as configuration files, data files, caches.

It would be nice to make `conda` search for the `condarc` file in `~/.config/conda/condarc` by just adding the path into the `SEARCH_PATH`.

```diff
SEARCH_PATH += (
    '$CONDA_ROOT/.condarc',
    '$CONDA_ROOT/condarc',
    '$CONDA_ROOT/condarc.d/',
+   '$XDG_CONFIG_HOME/conda/.condarc',
+   '$XDG_CONFIG_HOME/conda/condarc',
+   '$XDG_CONFIG_HOME/conda/condarc.d/',
+   '~/.config/conda/.condarc',
+   '~/.config/conda/condarc',
+   '~/.config/conda/condarc.d/',
    '~/.conda/.condarc',
    '~/.conda/condarc',
    '~/.conda/condarc.d/',
    '~/.condarc',
    '$CONDA_PREFIX/.condarc',
    '$CONDA_PREFIX/condarc',
    '$CONDA_PREFIX/condarc.d/',
    '$CONDARC',
)
```

Related issue: #9333 #10620.

### CODE REVIEW

File: conda/base/constants.py [hunk 1]
```diff
@@ -42,6 +42,12 @@
     )
 
 SEARCH_PATH += (
+    '$XDG_CONFIG_HOME/conda/.condarc',
+    '$XDG_CONFIG_HOME/conda/condarc',
+    '$XDG_CONFIG_HOME/conda/condarc.d/',
+    '~/.config/conda/.condarc',
+    '~/.config/conda/condarc',
+    '~/.config/conda/condarc.d/',
```
Comment (chenghlee) [inline conda/base/constants.py:50]: For consistency, I would suggest moving the XDG directories after `$CONDA_ROOT` and before `~/.conda`; that way, the install directory is always searched before $HOME and its subdirectories.

**Review by chenghlee — CHANGES_REQUESTED** [review]

**Comment by chenghlee** [comment]
@anaconda-issue-bot check

File: conda/base/constants.py (hunk 1 above)
Comment (Hsins) [reply conda/base/constants.py:53]: Done with that, I've moved the XDG directories after `$CONDA_ROOT` and before `~/.conda` for consistency.

**Review by jezdez — APPROVED** [review]
I think this is ready to go.

File: conda/base/constants.py (hunk 1 above)
Comment (chenghlee) [reply conda/base/constants.py:53]: Thanks for making the change. PR looks good to go now.

**Review by chenghlee — APPROVED** [review]
LGTM

**Pull request merged**